gocourse04/smart_zoo_test.go

```diff
@@ -0,0 +1,218 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+)
+import "github.com/stretchr/testify/assert"
+
+func TestAddSector(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	area.AddSector(sector)
+
+	s, exists := area.GetSector("sector_1")
+	assert.True(t, exists, "Sector should exist")
+	assert.Equal(t, sector, s, "Sectors should be equal")
+}
+
+func TestDeleteSector(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	area.AddSector(sector)
+
+	_, exists := area.GetSector("sector_1")
+	assert.True(t, exists, "Sector shouldn't exist")
+
+	area.DeleteSector("sector_1")
+	_, exists = area.GetSector("sector_1")
+	assert.False(t, exists, "Sector shouldn't exist")
+}
+
+func TestAddAnimal(t *testing.T) {
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	animal := NewAnimal("animal_1", "Frodo")
+	sector.AddAnimal(animal)
+
+	s, exists := sector.GetAnimal("animal_1")
+	assert.True(t, exists, "Animal should exist")
+	assert.Equal(t, animal, s, "Animals should be equal")
+}
+
+func TestDeleteAnimal(t *testing.T) {
+	sector := NewSector("sector_1", "Parrots", NewTechnicalRoom("tr_1"))
+	animal := NewAnimal("animal_1", "Frodo")
+	sector.AddAnimal(animal)
+
+	_, exists := sector.GetAnimal("animal_1")
+	assert.True(t, exists, "Animal should exist")
+
+	sector.DeleteAnimal("animal_1")
+	_, exists = sector.GetAnimal("animal_1")
+	assert.False(t, exists, "Animal shouldn't exist")
+}
+
+func TestAddArea(t *testing.T) {
+	zoo := NewZoo("Happy zoo")
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	zoo.AddArea(area)
+
+	z, exists := zoo.GetArea("area_1")
+	assert.True(t, exists, "Area should exist")
+	assert.Equal(t, area, z, "Areas should be equal")
+}
+
+func TestDeleteArea(t *testing.T) {
+	zoo := NewZoo("Happy zoo")
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	zoo.AddArea(area)
+
+	_, exists := zoo.GetArea("area_1")
+	assert.True(t, exists, "Area should exist")
+
+	zoo.DeleteArea("area_1")
+	_, exists = zoo.GetArea("area_1")
+	assert.False(t, exists, "Area shouldn't exist")
+}
+
+func TestLookupAnimalByID(t *testing.T) {
+	area := NewArea("area_1", "Happy birds", "Feathered")
+	sector := NewSector("sector_1", "Parrots", nil)
+	sector2 := NewSector("sector_2", "Ducks", nil)
+	animal := NewAnimal("animal_1", "Frodo")
+	animal2 := NewAnimal("animal_2", "Bilbo")
```

Comment:
Ось так легше читатиметься:
```suggestion
	parrotsSector := NewSector("sector_1", "Parrots", nil)
	ducksSector := NewSector("sector_2", "Ducks", nil)
	frodo := NewAnimal("animal_1", "Frodo")
	bilbo := NewAnimal("animal_2", "Bilbo")
```

Змінні типу `sector1, sector2, animal1, animal2, animal3` бажано уникати навіть в тестах. Тести - це по суті, той же код. І до нього такі ж стандарти якості застосовуються.

Reply:
В усіх інших функціях теж бажано прибрати цифри в назвах змінних.